app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/VolleyErrorHandler.java

```diff
@@ -0,0 +1,123 @@
+package fr.iut_rodez.pathpilot_android_client.util;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.android.volley.AuthFailureError;
+import com.android.volley.NetworkError;
+import com.android.volley.NoConnectionError;
+import com.android.volley.ParseError;
+import com.android.volley.ServerError;
+import com.android.volley.TimeoutError;
+import com.android.volley.VolleyError;
+
+import java.nio.charset.StandardCharsets;
+
+import fr.iut_rodez.pathpilot_android_client.R;
+
+public class VolleyErrorHandler {
+    private static final String TAG = "VolleyErrorHandler";
+
+    /**
+     * Handles Volley errors and provides user-friendly error messages
+     *
+     * @param context Application context
+     * @param error Volley error to handle
+     * @return User-friendly error message
+     */
+    public static String handleError(Context context, VolleyError error) {
+        String errorMessage = "Unknown error occurred";
+
+        // Check network connectivity first
+        if (!Network.isNetworkConnected(context)) {
+            return "No internet connection. Please check your network settings.";
+        }
+
+        // Detailed error handling
+        if (error instanceof NetworkError) {
+            errorMessage = "Network error. Unable to connect to the server.";
+        } else if (error instanceof ServerError) {
+            errorMessage = handleServerError(error);
+        } else if (error instanceof AuthFailureError) {
+            errorMessage = "Authentication failed. Please give valid credentials.";
+        } else if (error instanceof ParseError) {
+            errorMessage = "Unable to process server response.";
+        } else if (error instanceof TimeoutError) {
+            errorMessage = "Connection timed out. Please try again.";
+        }
+
+        // Log detailed error information
+        logVolleyError(error);
+
+        // Show toast notification
+        Popup popup = new Popup(context);
+        popup.showAlertDialog(context.getString(R.string.error), errorMessage);
+
+        return errorMessage;
+    }
+
+    /**
+     * Handles server-specific error messages
+     *
+     * @param error Volley error
+     * @return Specific server error message
+     */
+    private static String handleServerError(VolleyError error) {
```

Comment:
Same here with enhanced switch like this 

```java
return switch (day) {
            case MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY -> "Weekday";
            case SATURDAY, SUNDAY -> "Weekend";
        };
```

Reply:
Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/38/commits/aabf7d0cb168a6b98d443e23b957c88983e1d813